static/panes/flags-view.js

```diff
@@ -0,0 +1,208 @@
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+'use strict';
+
+var FontScale = require('../fontscale');
+var monaco = require('monaco-editor');
+var _ = require('underscore');
+var $ = require('jquery');
+var ga = require('../analytics');
+var monacoConfig = require('../monaco-config');
+var local = require('../local');
+
+function Flags(hub, container, state) {
+    state = state || {};
+    this.container = container;
+    this.eventHub = hub.createEventHub();
+    this.domRoot = container.getElement();
+    this.domRoot.html($('#flags').html());
+    this.source = state.source || '';
+    this._currentDecorations = [];
+    var root = this.domRoot.find('.monaco-placeholder');
+
+    this.settings = JSON.parse(local.get('settings', '{}'));
+
+    this.editor = monaco.editor.create(root[0], monacoConfig.extendConfig({
+        value: state.compilerFlags.replaceAll(' ', '\n'),
+        language: 'plaintext',
+        readOnly: false,
+        glyphMargin: true,
+    }));
+
+    this._compilerid = state.id;
+    this._compilerName = state.compilerName;
+    this._editorid = state.editorid;
+
+    this.awaitingInitialResults = false;
+    this.selection = state.selection;
+
+    this.isCompilerSupported = false;
+
+    this.initButtons(state);
+    this.initCallbacks();
+
+    this.setTitle();
+    this.onSettingsChange(this.settings);
+    this.eventHub.emit('flagsViewOpened', this._compilerid);
+    ga.proxy('send', {
+        hitType: 'event',
+        eventCategory: 'OpenViewPane',
+        eventAction: 'detailedCompilerFlags',
+    });
+}
+
+Flags.prototype.initButtons = function (state) {
+    this.fontScale = new FontScale(this.domRoot, state, this.editor);
+
+    this.topBar = this.domRoot.find('.top-bar');
+};
+
+Flags.prototype.initCallbacks = function () {
+    this.fontScale.on('change', _.bind(this.updateState, this));
+
+    this.eventHub.on('compiler', this.onCompiler, this);
+    this.eventHub.on('compilerClose', this.onCompilerClose, this);
+    this.eventHub.on('settingsChange', this.onSettingsChange, this);
+    this.eventHub.on('resize', this.resize, this);
+    this.container.on('destroy', this.close, this);
+    this.eventHub.emit('requestSettings');
+    this.eventHub.emit('findCompilers');
+
+    this.container.on('resize', this.resize, this);
+    this.container.on('shown', this.resize, this);
+
+    this.container.layoutManager.on('initialised', function () {
+        // Once initialized, let everyone know what text we have.
+        this.maybeEmitChange();
+    }, this);
+    this.eventHub.on('initialised', this.maybeEmitChange, this);
+
+    this.editor.getModel().onDidChangeContent(_.bind(function () {
+        this.debouncedEmitChange();
+        this.updateState();
+    }, this));
+
+    this.cursorSelectionThrottledFunction =
```

Comment:
Is this used anywhere? I think it's copy/pasted?

Reply:
It's definitely copy/pasted. Most views seem to have this; but I can't figure out where it's called or what it does. It seems like either it's needed in all of them, or none of them.  (Or I'm wrong!)  Let me know what you think...

Reply:
This is used so that Selections in the editors are persisted across sessions. 
This is helpful so that, on page load, a certain line in every editor is shown, instead of the top of the text, fi a selection was made (Check how you're shown line 90 here https://godbolt.org/z/84375aMGq)
So it's useful in every pane that contains a monaco editor, so I'd leave this here